src/pipx/pipx_metadata_file.py

```diff
@@ -35,11 +35,14 @@ class PackageInfo(NamedTuple):
     app_paths_of_dependencies: Dict[str, List[Path]]
     package_version: str
     suffix: str = ""
+    # whether the package was selected with `pipx select`
+    selected_as_default: bool = False
 
 
 class PipxMetadata:
     # Only change this if file format changes
-    __METADATA_VERSION__: str = "0.2"
+    # This is also necessary when adding a field that has a default value
+    __METADATA_VERSION__: str = "0.3"
```

Comment:
When you modify the metadata you need to update the code to gracefully recover from old metadata.  The immediate task is to modify `_convert_legacy_metadata()` to convert from all previous versions to the newest version.  Likely this will just mean adding the default value to the read json for `selected_as_default=False`.

Reply:
Should this not be automatically populated because I set a default value [here](https://github.com/pipxproject/pipx/pull/535/files#diff-902c1a727ef195654d9382fd98ccae279cf6eaad86f72621c3e8a2fe28d15d30R39)?

Reply:
I think that should be correct.  But you still need to modify `_convert_legacy_metadata()` so that it doesn't raise a PipxError if it receives `pipx_metadata_version==0.2`.  And double-check that the other versions are still doing the proper thing (which I think they will.)